tests/Secrets/tst_secretsrequests/tst_secretsrequests.cpp

```diff
@@ -275,7 +275,11 @@ void tst_secretsrequests::devicelockCollectionSecret()
     QCOMPARE(gsrss.count(), 2);
     QCOMPARE(gsr.status(), Request::Finished);
     QCOMPARE(gsr.result().code(), Result::Succeeded);
-    QCOMPARE(gsr.secret(), testSecret);
+    QCOMPARE(gsr.secret().data(), testSecret.data());
+    QCOMPARE(gsr.secret().type(), testSecret.type());
+    QCOMPARE(gsr.secret().filterData(), testSecret.filterData());
+    QCOMPARE(gsr.secret().name(), testSecret.name());
+    QCOMPARE(gsr.secret().collectionName(), testSecret.collectionName());
```

Comment:
This is a bit unfortunate. At least create shared function to compare secrets?

Reply:
Sure, let's do that - after the branch.  This should not block this PR.